DSharpPlus.Commands/Converters/DiscordAttachmentConverter.cs

```diff
@@ -15,54 +16,29 @@ public class AttachmentConverter : ISlashArgumentConverter<DiscordAttachment>, I
     public string ReadableName => "Discord File";
     public bool RequiresText => false;
 
-    public Task<Optional<DiscordAttachment>> ConvertAsync(TextConverterContext context, MessageCreatedEventArgs eventArgs)
+    public Task<Optional<DiscordAttachment>> ConvertAsync(ConverterContext context)
     {
-        DiscordMessage message = eventArgs.Message;
-        int currentAttachmentArgumentIndex = context.Command.Parameters.Where(argument => argument.Type == typeof(DiscordAttachment)).IndexOf(context.Parameter);
-        if (context.Parameter.Attributes.FirstOrDefault(x => x is TextMessageReplyAttribute) is TextMessageReplyAttribute textMessageReplyAttribute)
-        {
-            if (eventArgs.Message.ReferencedMessage is not null)
-            {
-                message = eventArgs.Message.ReferencedMessage;
-            }
-            else if (textMessageReplyAttribute.RequireReplies)
-            {
-                // No referenced message, but the attribute requires it
-                return Task.FromResult(Optional.FromNoValue<DiscordAttachment>());
-            }
-        }
-
-        return message.Attachments.Count <= currentAttachmentArgumentIndex
-            ? Task.FromResult(Optional.FromNoValue<DiscordAttachment>())
-            : Task.FromResult(Optional.FromValue(message.Attachments[currentAttachmentArgumentIndex]));
-    }
+        IReadOnlyList<CommandParameter> attachmentParameters = context.Command.Parameters.Where(argument => argument.Type == typeof(DiscordAttachment)).ToList();
+        int currentAttachmentArgumentIndex = attachmentParameters.IndexOf(context.Parameter);

```

Comment:
I suspect this will break when a user does, say, `DiscordAttachment[]` - I'm sure this is something that Aki's command tree/metadata changes could solve.

Reply:
Brilliant catch. I think there should be a fix for this, I'll think on it for a bit.